### PR TITLE
Setup file structure and cmake for the core and engine API libraries

### PR DESCRIPTION
**Pull request by EdSwordsmith**
# Description

Creates a new directory `api/` with two projects for the core and engine C API libraries. Both have their own CMakeLists.txt file and a samples directory which will probably be needed later for testing these new libraries.

## Checklist

- [x] Self-review changes.
- [ ] Evaluate impact on the documentation.
- [ ] Ensure test coverage.
- [ ] Write new samples.
- [ ] Add entry to the changelog's unreleased section.

### CODE REVIEW

File: api/core/include/cubos/api/core/api.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/// @file
+/// @brief Macro @ref CUBOS_CORE_C_API.
+/// @ingroup core
```
Comment (RiscadoA) [inline api/core/include/cubos/api/core/api.h:3]:
```suggestion
/// @ingroup core-api
```

File: api/core/include/cubos/api/core/api.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+#pragma once
+
+/// @brief Macro used to export and import symbols from the library.
+/// @addtogroup core
```
Comment (RiscadoA) [inline api/core/include/cubos/api/core/api.h:8]:
```suggestion
/// @addtogroup core-api
```

File: api/engine/include/cubos/api/engine/api.h [hunk 3]
```diff
@@ -0,0 +1,23 @@
+/// @file
+/// @brief Macro @ref CUBOS_ENGINE_C_API.
+/// @ingroup engine
```
Comment (RiscadoA) [inline api/engine/include/cubos/api/engine/api.h:3]:
```suggestion
/// @ingroup engine-api
```

File: api/engine/include/cubos/api/engine/api.h [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+#pragma once
+
+/// @brief Macro used to export and import symbols from the library.
+/// @addtogroup engine
```
Comment (RiscadoA) [inline api/engine/include/cubos/api/engine/api.h:8]:
```suggestion
/// @addtogroup engine-api
```

File: api/core/samples/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# api/core/samples/CMakeLists.txt
+# Core C API samples build configuration
+
```
Comment (RiscadoA) [inline api/core/samples/CMakeLists.txt:3]:
```suggestion
```

File: api/engine/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,30 @@
+set(CUBOS_API_ENGINE_SOURCE
+    "src/api.cpp"
+)
+
```
Comment (RiscadoA) [inline api/engine/CMakeLists.txt:4]:
```suggestion
```

**Review by RiscadoA — CHANGES_REQUESTED** [review]
Thanks for working on this :rocket: 
Could you also add a `module.dox` file to both the `include/cubos/api/core` and `include/cubos/api/engine` directories? You can just copy the ones in the non-api directories, and change their names (to core-api, engine-api) and descriptions to match them.
Forgot documentation was thing

**Review by RiscadoA — CHANGES_REQUESTED** [review]
One more thing: you should also add the new directories to the `docs/Doxyfile` directory, otherwise they won't be included in the generated docs.

**Review by RiscadoA — APPROVED** [review]
Other than the comments I just added, LGTM!

**Review by teres4 — APPROVED** [review]

**Pull request merged**